lib/board.js

```diff
@@ -53,99 +53,56 @@ function setupNodeGrid(nodes){
 
 function setNodeRelationships(nodes) {
   nodes.forEach(function(node) {
-    setUpNode(node, nodes);
-    setDownNode(node, nodes);
-    setLeftNode(node, nodes);
-    setRightNode(node, nodes);
+    assignNeighbors(node, nodes);
   });
 }
 
-function setUpNode(node, nodes) {
-  var topIds = ["1", "2", "3", "4"];
-  if (topIds.indexOf(node.id) > -1 ) {
-    node.upNode = "edge";
-  } else {
-    var currentNodeIndex = nodes.indexOf(node);
-    node.upNode = nodes[currentNodeIndex - 4];
-  }
-}
+var edgeIds = {"up": ["1", "2", "3", "4"],
+               "down": ["13", "14", "15", "16"],
+               "left": ["1", "5", "9", "13"],
+               "right": ["4", "8", "12", "16"]};
 
-function setDownNode(node, nodes) {
-  var bottomIds = ["13", "14", "15", "16"];
-  if (bottomIds.indexOf(node.id) > -1 ) {
-    node.downNode = "edge";
-  } else {
-    var currentNodeIndex = nodes.indexOf(node);
-    node.downNode = nodes[currentNodeIndex + 4];
-  }
+function assignNeighbors(node, nodes) {
+  ["up", "down", "left", "right"].forEach(function(direction) {
+    if (edgeIds[direction].indexOf(node.id) > -1) {
+      setEdge(node, direction);
+    } else {
+      var currentNodeIndex = nodes.indexOf(node);
+      if (direction === "up") {
+        node.upNode = nodes[currentNodeIndex - 4];
+      } else if (direction === "down") {
+        node.downNode = nodes[currentNodeIndex + 4];
+      } else if (direction === "left") {
+        node.leftNode = nodes[currentNodeIndex - 1];
+      } else if (direction === "right") {
+        node.rightNode = nodes[currentNodeIndex + 1];
+      }
+    }
+  });
 }
 
-function setLeftNode(node, nodes) {
-  var leftIds = ["1", "5", "9", "13"];
-  if (leftIds.indexOf(node.id) > -1 ) {
+function setEdge(node, direction) {
+  if (direction === "up") {
+    node.upNode = "edge";
+  } else if (direction === "down") {
+    node.downNode = "edge";
+  } else if (direction === "left") {
     node.leftNode = "edge";
-  } else {
-    var currentNodeIndex = nodes.indexOf(node);
-    node.leftNode = nodes[currentNodeIndex - 1];
-  }
-}
-
-function setRightNode(node, nodes) {
-  var leftIds = ["4", "8", "12", "16"];
-  if (leftIds.indexOf(node.id) > -1 ) {
+  } else if (direction === "right") {
     node.rightNode = "edge";
-  } else {
-    var currentNodeIndex = nodes.indexOf(node);
-    node.rightNode = nodes[currentNodeIndex + 1];
   }
 }
 
-function createUpLists(nodes) {
```

Comment:
Thanks! Similar opportunities appear throughout our code so I was glad to get a start on it.  I tried looking into some metaprogramming-ish but had no luck.  Thought this was a decent start.